app/src/main/java/explorer/city/com/cityexplorer/network/CacheControlInterceptor.kt

```diff
@@ -0,0 +1,25 @@
+package explorer.city.com.cityexplorer.network
+
+import explorer.city.com.cityexplorer.BaseApplication
+import okhttp3.Interceptor
+import okhttp3.Response
+
+class CacheControlInterceptor : Interceptor {
+    override fun intercept(chain: Interceptor.Chain?): Response {
+        val originalResponse = chain?.proceed(chain.request())
+        if (BaseApplication.instance.isNetworkAvailable()) {
+            val maxAge = 60
+            return originalResponse
+                    ?.newBuilder()
+                    ?.header("Cache-Control", "public, max-age=" + maxAge)
+                    ?.build()!!
```

Comment:
I have read that if you use `!!` you are doing something wrong. But if you know that it's not null, it's fine